test_libs/pyspec/eth2spec/test/context.py

```diff
@@ -80,3 +88,35 @@ def entry(*args, **kw):
         bls.bls_active = old_state
         return out
     return entry
+
+
+def with_phase0(fn):
+    """
+    Decorator to use phase 0's spec.
+    """
+    def entry(*args, **kw):
+        kw['spec'] = spec_phase0
+        return fn(*args, **kw)
+    return entry
+
+
+def with_phase1(fn):
+    """
+    Decorator to use phase 1's spec
+    """
+    def entry(*args, **kw):
+        kw['spec'] = spec_phase1
+        return fn(*args, **kw)
+    return entry
+
+
+def with_all_phases(fn):
```

Comment:
I like this idea and it may be a good way of resolving the state generator issue w.r.t. only running the appropriate phase.

Reply:
I swapped out the decorators for parameterised decorator factories so the behaviour you described is now the default. :)

Reply:
whoops! I'm not certain I was recommending to change the default from `@with_all_phases`. I think `with_all_phases` is a better default because we assume functionality works when we add a new phase unless we explicitly make it not work. I just thought it might be useful when distinguishing from the default `@with_phases(['phase0'])` rather than `with_phase0`.

I guess I want the default behavior to have to explicitly pull tests out for a phase rather than have to put tests in for a phase. Thoughts? Sorry for my miscommunication in my vague comment

Reply:
I had this debate myself. I'll re-add `with_all_phases`. It'll basically be an alias for the `with_phases` decorator factory, but with all possible phases populating the list.

I could also add an opt out decorator. Something like `with_all_phases_except(['phase0'])`. What are your thoughts?

Reply:
Yeah, that would likely work well for things that are `>=` some phase. Most of the phase1 will be "all phases except 0"

Reply:
👍 I implemented in this in the [latest commit.](https://github.com/ethereum/eth2.0-specs/pull/1061/commits/67471a8d6e802115513c4014fa25c6abd28c7f35)